pinot-core/src/test/java/org/apache/pinot/queries/FilteredAggregationsTest.java

```diff
@@ -131,13 +140,7 @@ private void buildSegment(String segmentName)
 
     TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName(TABLE_NAME)
         .setInvertedIndexColumns(Collections.singletonList(INT_COL_NAME)).setFieldConfigList(fieldConfigs).build();
-    Schema schema = new Schema.SchemaBuilder().setSchemaName(TABLE_NAME)
-        .addSingleValueDimension(NO_INDEX_INT_COL_NAME, FieldSpec.DataType.INT)
-        .addSingleValueDimension(STATIC_INT_COL_NAME, FieldSpec.DataType.INT)
-        .addSingleValueDimension(BOOLEAN_COL_NAME, FieldSpec.DataType.BOOLEAN)
-        .addSingleValueDimension(STRING_COL_NAME, FieldSpec.DataType.STRING)
-        .addMetric(INT_COL_NAME, FieldSpec.DataType.INT).build();
-    SegmentGeneratorConfig config = new SegmentGeneratorConfig(tableConfig, schema);
+    SegmentGeneratorConfig config = new SegmentGeneratorConfig(tableConfig, SCHEMA);
```

Comment:
Reading the test again, actually we are not testing reloading here. In that case we can directly create a final table config and keep it as constant similar to schema. We should use the same table config to create and load the segment

Reply:
Yup, thanks for pointing it